src/background/transactions/TransactionService.ts

```diff
@@ -52,9 +67,88 @@ export class TransactionService {
 
   async initialize() {
     await this.transactionsStore.ready();
-    const pending = getPendingTransactions(this.transactionsStore.getState());
+    const transactions = this.transactionsStore.getState();
+    const pending = getPendingTransactions(transactions);
     this.transactionsPoller.add(pending.map(toPollingObj));
     this.addListeners();
+    if (transactions.length) {
+      this.startPurgeInterval({ leading: true });
+    }
+  }
+
+  private startPurgeInterval({ leading } = { leading: false }) {
+    if (leading) {
+      this.performPurgeCheck(); // make leading call
+    }
+    this.purgeInterval.start(ONE_DAY);
+  }
+
+  private schedulePurgeCheck = throttle(
+    () => {
+      this.performPurgeCheck();
+    },
+    FOUR_MINUTES,
+    { leading: false } // Invoke no sooner and no more frequent than FOUR_MINUTES
+  );
+
+  /**
+   * Purges from cache all transactions belonging to {address} in {chainId}
+   * which have a nonce less than or equals to {fromNonce}
+   */
+  private async purgeEntries({
+    address,
+    fromNonce,
+    chainId,
+  }: {
+    address: string;
+    fromNonce: number;
+    chainId: ChainId;
+  }) {
+    const transactions = await this.transactionsStore.getSavedState();
+    const candidates = transactions.filter((item) => {
```

Comment:
This is indeed an optimisation, but it looks like we can pass all transactions to `bulkDeleteTransactionsByHash` without significant difference in speed

Reply:
Incorrect comment